libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -957,6 +957,23 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			);
 			break;
 		}
+		case FunctionType::Kind::Wrap:
+		case FunctionType::Kind::Unwrap:
+		{
+			solAssert(arguments.size() == 1, "");
+			Type const* argumentType = arguments.at(0)->annotation().type;
+			solAssert(argumentType, "");
+			FunctionType::Kind kind = functionType->kind();
+			auto userDefinedType = kind == FunctionType::Kind::Wrap ?
+				dynamic_cast<UserDefinedValueType const*>(_functionCall.annotation().type) :
+				dynamic_cast<UserDefinedValueType const*>(argumentType);
+			solAssert(userDefinedType, "");
+			if (kind == FunctionType::Kind::Wrap)
+				solAssert(argumentType->isImplicitlyConvertibleTo(userDefinedType->underlyingType()), "");
+
+			acceptAndConvert(*arguments[0], *function.parameterTypes()[0], true);
```

Comment:
Do we really need to do cleanup?

Reply:
Can you assert that the implicit conversion is possible, please?

Reply:
I've removed the cleanup and added one more assert.